chain/cosmos/broadcaster.go

```diff
@@ -141,6 +141,16 @@ func (b *Broadcaster) UnmarshalTxResponseBytes(ctx context.Context, bytes []byte
 	if err := b.chain.cfg.EncodingConfig.Codec.UnmarshalJSON(bytes, &resp); err != nil {
 		return sdk.TxResponse{}, err
 	}
+
+	// persist nested errors such as ValidateBasic checks.
+	code := resp.Code
+	rawLog := resp.RawLog
+
+	// rawLog can be empty or just an empty array "[]"
+	if code != 0 || len(rawLog) > 2 {
+		return resp, fmt.Errorf("error in transaction (code: %d): %s", code, rawLog)
+	}
```

Comment:
why is `len(rawLog) > 2` indicative of an error if `code == 0`?

Reply:
I was thinking for ValidateBasic(), but ictest catches this in the Simulation (in BroadcastTx) vs at broadcast time. So should never be the base

Good call

Reply:
updated: 671f3ed1bd15313dff7fd2da2f8c7b5fe20d2a23